fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergApiConverter.java

```diff
@@ -129,10 +140,54 @@ public static Schema toIcebergApiSchema(List<Column> columns) {
     }
 
     // TODO(stephen): support iceberg transform partition like `partition by day(dt)`
-    public static PartitionSpec parsePartitionFields(Schema schema, List<String> fields) {
+    public static PartitionSpec parsePartitionFields(Schema schema, ListPartitionDesc desc) {
         PartitionSpec.Builder builder = PartitionSpec.builderFor(schema);
-        for (String field : fields) {
-            builder.identity(field);
+        if (desc == null) {
+            return builder.build();
+        }
+        int idx = 0;
+        for (String colName : desc.getPartitionColNames()) {
+            if (colName.startsWith(FeConstants.GENERATED_PARTITION_COLUMN_PREFIX)) {
```

Comment:
use `if (!...) continue` will make code less indented,  and easier to read (personal coding tyle only).